sources/mongodb/helpers.py

```diff
@@ -238,6 +325,29 @@ def convert_mongo_objs(value: Any) -> Any:
     return value
 
 
+def convert_arrow_columns(table: Any) -> Any:
+    """Convert the given table columns to Python types.
+
+    Args:
+        table (pyarrow.lib.Table): The table to convert.
+
+    Returns:
+        pyarrow.lib.Table: The table with the columns converted.
+    """
+    import pymongoarrow  # type: ignore
+    from dlt.common.libs.pyarrow import pyarrow
+
+    for i, field in enumerate(table.schema):
+        if pymongoarrow.types._is_objectid(field.type):
+            col_values = [str(value) for value in table[field.name]]
```

Comment:
OK I see we'll not be able to convert to hex in arrow. Still it would be good to avoid creating python lists and loops.

you could use numpy.vectorize with `hex` function. you may need to convert to binary first and then go to numpy array